src/ts/components/adclickoverlay.ts

```diff
@@ -10,39 +10,30 @@ export class AdClickOverlay extends ClickOverlay {
   configure(player: PlayerAPI, uimanager: UIInstanceManager): void {
     super.configure(player, uimanager);
 
-    let clickThroughUrl = <string>null;
-    let clickThroughEnabled = !player.getConfig().advertising
-      || !player.getConfig().advertising.hasOwnProperty('clickThroughEnabled')
-      || (player.getConfig().advertising as any).clickThroughEnabled;
+    let clickThroughCallback: () => void = null;
 
     player.on(player.exports.PlayerEvent.AdStarted, (event: AdEvent) => {
-      clickThroughUrl = null; // TODO event.clickThroughUrl;
-
-      if (clickThroughEnabled) {
-        this.setUrl(clickThroughUrl);
-      } else {
-        // If click-through is disabled, we set the url to null to avoid it open
-        this.setUrl(null);
-      }
+      let ad = event.ad;
+      this.setUrl(ad.clickThroughUrl);
+      clickThroughCallback = ad.adClickedCallback;
     });
 
     // Clear click-through URL when ad has finished
     let adFinishedHandler = () => {
       this.setUrl(null);
     };
+
     player.on(player.exports.PlayerEvent.AdFinished, adFinishedHandler);
     player.on(player.exports.PlayerEvent.AdSkipped, adFinishedHandler);
     player.on(player.exports.PlayerEvent.AdError, adFinishedHandler);
 
     this.onClick.subscribe(() => {
       // Pause the ad when overlay is clicked
-      player.pause('ui-content-click');
+      player.pause('ui-ad-click-overlay');
```

Comment:
do we have to pause the content here? 

Also we should not add the `ad` name, because they might be detected more easily by ad blockers.

Reply:
Yes when the clickthough Url is opnened the ad needs to be paused.

Reply:
If you're referring to last week's discussion, ad blockers are looking for "ad" in DOM classes and IDs, and maybe filenames, but most certainly not values of random variables in JS code. If that were the case, our whole UI would have alwas been blocked (`AdClickOverlay`, `AdSkipButton`, etc).